bokeh/document.py

```diff
@@ -298,6 +300,12 @@ def __init__(self, **kwargs):
         self._callbacks = {}
         self._session_callbacks = {}
         self._session_context = None
+        self._modules = []
+
+    def __del__(self):
+        for module in self._modules:
+            if module.__name__ in sys.modules:
+                del sys.modules[module.__name__]
```

Comment:
`modify_document` is called for every new session, its what generates the document for the session.